### PR TITLE
fix(tests): remove hard-coding + wait for responses

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

This should not impact local tests/CI, but makes it possible to test against remote instances of the client. While we may not want to do this long term, it is quite useful right now.

### CODE REVIEW

File: e2e/utils/add-growthbook-cookie.ts [hunk 1]
```diff
@@ -11,7 +11,9 @@ export async function addGrowthbookCookie({
     {
       name: 'gbuuid',
       value: variation,
-      domain: 'localhost',
+      // Weirdly, it seems necessary to prefix the domain with a dot. This seems
+      // to be a peculiarity of Playwright.
+      domain: process.env.COOKIE_DOMAIN || 'localhost',
```
Comment (ShaunSHamilton) [inline e2e/utils/add-growthbook-cookie.ts:16]: I do not see a _dot_?

Comment (ojeytonwilliams) [reply e2e/utils/add-growthbook-cookie.ts:16]:
```suggestion
      // Weirdly, it seems necessary to prefix the domain in COOKIE_DOMAIN i.e.
      // if the domain is freecodecamp.org then the COOKIE_DOMAIN must be
      // .freecodecamp.org. This seems to be a peculiarity of Playwright.
      domain: process.env.COOKIE_DOMAIN || 'localhost',
```
Yeah, that could have been clearer. How about this?

Comment (ShaunSHamilton) [reply e2e/utils/add-growthbook-cookie.ts:16]: Does that mean this needs to change?

https://github.com/freeCodeCamp/freeCodeCamp/blob/43661cf867796037f93f1a7fb640f94b2cd1d1ea/.gitpod.yml#L33

Comment (ojeytonwilliams) [reply e2e/utils/add-growthbook-cookie.ts:16]: Good catch. I'll try changing that and see if anything breaks (it shouldn't).

Comment (ojeytonwilliams) [reply e2e/utils/add-growthbook-cookie.ts:16]: Seems fine.

**Review by ShaunSHamilton — APPROVED** [review]
LGTM 👍

**Review by Sembauke — APPROVED** [review]

**Pull request merged**